master/internal/api_resourcepool_intg_test.go

```diff
@@ -0,0 +1,366 @@
+//go:build integration
+// +build integration
+
+package internal
+
+import (
+	"context"
+	"testing"
+
+	"github.com/uptrace/bun"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/mocks"
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/resourcepoolv1"
+)
+
+const (
+	testPoolName       = "testRP"
+	testPool2Name      = "testRP2"
+	testWorkspaceName  = "bindings_test_workspace_1"
+	testWorkspace2Name = "bindings_test_workspace_2"
+)
+
+func setupWorkspaces(ctx context.Context, t *testing.T, api *apiServer) []int32 {
+	w1, err := api.PostWorkspace(ctx, &apiv1.PostWorkspaceRequest{Name: testWorkspaceName})
+	require.NoError(t, err)
+	w2, err := api.PostWorkspace(ctx, &apiv1.PostWorkspaceRequest{Name: testWorkspace2Name})
+	require.NoError(t, err)
+
+	return []int32{w1.Workspace.Id, w2.Workspace.Id}
+}
+
+func cleanupWorkspaces(ctx context.Context) {
```

Comment:
You may have already consider this and it is awkward to implement. 

It would be nice if we can use api.DeleteWorkspace() here.